File: member_crud-jpa/app/src/main/java/sample/member/app/web/member/MemberViewController.java
package sample.member.app.web.member;

import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import sample.member.app.security.SimpleUser;

@Controller
@RequestMapping("/member")
public class MemberViewController {

	@RequestMapping(value = "", method = RequestMethod.GET)
	public String list() {
		return "member-list.html";
	}

	@RequestMapping(value = "/my", method = RequestMethod.GET)
	public String my(Model model) {
		SimpleUser simpleUser = (SimpleUser) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
		model.addAttribute("username", simpleUser.getUsername());
		return "member-detail.html";
	}

	@RequestMapping(value = "/{username}", method = RequestMethod.GET)
	public String detail(@PathVariable String username, Model model) {
		model.addAttribute("username", username);
		return "member-detail.html";
	}

}

File: 2022-webmvc/app/member/src/main/kotlin/sample/app/web/member/MemberViewController.kt
// package sample.app.web.member
//
// import org.springframework.security.core.context.SecurityContextHolder
// import org.springframework.stereotype.Controller
// import org.springframework.ui.Model
// import org.springframework.web.bind.annotation.PathVariable
// import org.springframework.web.bind.annotation.RequestMapping
// import org.springframework.web.bind.annotation.RequestMethod
// import sample.app.security.SimpleUser
//
// @Controller
// @RequestMapping("/member")
// class MemberViewController {
//
//    @RequestMapping(value = [""], method = [RequestMethod.GET])
//    fun list(): String = "member-list.html"
//
//    @RequestMapping(value = ["/my"], method = [RequestMethod.GET])
//    fun my(model: Model): String {
//        val simpleUser = SecurityContextHolder.getContext().authentication.principal as SimpleUser
//        model.addAttribute("username", simpleUser.username)
//        return "member-detail.html"
//    }
//
//    @RequestMapping(value = ["/{username}"], method = [RequestMethod.GET])
//    fun detail(@PathVariable username: String, model: Model): String {
//        model.addAttribute("username", username)
//        return "member-detail.html"
//    }
// }
